### PR TITLE
Add to object function

### PR DESCRIPTION
**Pull request by gregorybarale**

### CODE REVIEW

File: src/toObject/toObject.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * ```
+ * @example Using the chain
+ * ```
+ * chain([['a', 1], ['b', 2]])
+ *   .chain(toObject<(string | number)[][]>(item => item[0], item => item[1]))
+ *   .value() // {a: 1, b: 'c'}
+ * ```
+ */
+export function toObject<T, U>(
+  keyGetter: Getter<T, string>,
```
Comment (evanliomain) [inline src/toObject/toObject.ts]: No need to define a `Getter` type. Just use `(x: T) => string` and `(x: T) => U`

File: src/toObject/toObject.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  valueGetter: Getter<T, U>
+): (array: T[]) => { [key: string]: U } {
```
Comment (evanliomain) [inline src/toObject/toObject.ts]: Use `ReadonlyArray<T>` instead of `T[]`

File: src/toObject/toObject.spec.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+describe('toObject function', () => {
+  const input = [['a', 1], ['b', 'c']];
+
+  test('is a pure function', () => {
+    toObject(x => `${x}`, x => x)(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test("[['a', 1], ['b', 'c']] |> toObject(key => key, value => value + value) === {a: 1, b: 'c'}", () => {
+    expect(toObject(item => item[0], item => item[1])(input)).toEqual({
```
Comment (evanliomain) [inline src/toObject/toObject.spec.ts]: Use `.toMatchSnapshot();`

File: src/toObject/toObject.ts (hunk 1 above)
Comment (gregorybarale) [reply src/toObject/toObject.ts]: Done

File: src/toObject/toObject.ts (hunk 2 above)
Comment (gregorybarale) [reply src/toObject/toObject.ts]: Done

File: src/toObject/toObject.spec.ts (hunk 3 above)
Comment (gregorybarale) [reply src/toObject/toObject.spec.ts]: Done

File: src/toObject/toObject.spec.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+describe('toObject function', () => {
+  const input = [['a', 1], ['b', 'c']];
+
+  test('is a pure function', () => {
+    toObject(x => `${x}`, x => x)(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test("[['a', 1], ['b', 'c']] |> toObject(key => key, value => value + value) === {a: 1, b: 'c'}", () => {
```
Comment (evanliomain) [inline src/toObject/toObject.spec.ts]: `toObject(key => key, value => value + value)` functions not describes what the test did.

Comment (gregorybarale) [reply src/toObject/toObject.spec.ts]: Done

**Pull request merged**